src/GraphQl/Serializer/ItemNormalizer.php

```diff
@@ -57,16 +66,35 @@ protected function normalizeCollectionOfRelations(PropertyMetadata $propertyMeta
     /**
      * {@inheritdoc}
      */
-    public function supportsNormalization($data, $format = null)
+    public function supportsDenormalization($data, $type, $format = null)
     {
-        return self::FORMAT === $format && parent::supportsNormalization($data, $format);
+        return self::FORMAT === $format && parent::supportsDenormalization($data, $type, $format);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function supportsDenormalization($data, $type, $format = null)
+    protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
     {
-        return self::FORMAT === $format && parent::supportsDenormalization($data, $type, $format);
+        $allowedAttributes = parent::getAllowedAttributes($classOrObject, $context, $attributesAsString);
+
+        if (($context['api_denormalize'] ?? false) && false !== ($indexId = array_search('id', $allowedAttributes, true))) {
+            $allowedAttributes[] = '_id';
+            array_splice($allowedAttributes, $indexId, 1);
```

Comment:
Wouldn't `unset($allowedAttributes[$indexId]` be explicit and faster?

Reply:
Yes but I don't really like it, because it doesn't preserve keys and can lead to weird bugs.